cloudinit/config/schema.py

```diff
@@ -263,8 +281,16 @@ def _anyOf(
         if not errs:
             all_deprecations.extend(deprecations)
             break
+        if "type" in instance and "anyOf_type" in subschema.get("$ref", ""):
+            if f"anyOf_type_{instance['type']}" in subschema["$ref"]:
+                # A matching anyOf_type_XXX $ref indicates this is likely the
```

Comment:
Why is this likely the best match? Did something about network validation make this change, or is this just an enchancement? I'm having a hard time grokking this function in general, but that's just my lack of understand of json schema.

Reply:
Really this is a shortfall of our abundant use of anyOf conditions as it is hard to report 'the right' error in many cases. It's also a shortfall of network version 1 definitions which allowed 6 different potential objects in the same list differentiated only by a nested `type:` key within the dict. So, our JSON schema is now trying to cleanly represent an all too flexible schema w/ `anyOf` clauses which reduce our JSONschema's effectiveness in determining specific errors.

JSON schema's `best_match` algorithm which only says the [deepest match into the match path `anyOf` structures is likely the best match[(https://python-jsonschema.readthedocs.io/en/latest/errors/#best-match-and-relevance). But, since our `anyOf` subschemas for `type: (physical|bridge|bond|vlan)` all that happen to fail at the same depth, JSON schema's `best_match` algorithm chooses the wrong (first) item to report "most likely" errors from. 

This additional logic is an enhancement to the best match algorithm, because our instance["type"] is (bond|bridge|physical .. .etc) and our defined schema `$ref` name  happens to have encoded the specific type in the subschema name. If our current `instance['type']` == `bridge` and  `subschema[$ref] == anyOf_type_bridge`, we know errors in this subschema are the best matched errors for this object (as opposed to schema errors for `anyOf_type_vlan` which would likely just be `type is 'bridge' is not  equal to 'vlan'` among other vlan-specific required properties).

To give us more of a hint into the network types, I added a schema naming construct, which definitely needs more informative comments. Any schema reference name containing the prefix `anyOf_type_<type_name>` is a hint to our error matching logic that the subschema is intended to match `type: <type_name>`  objects.

Reply:
Very helpful, thanks! Any chance most of that comment could make its way into the docstring? I have a feeling I'm gonna be staring at this code in 6 months wondering what I'm looking at.

Reply:
done.